README.md

```diff
@@ -89,16 +89,179 @@ jobs:
 
       - name: Set up Mulesoft environment
-        uses: nimblehq/mulesoft-actions/setup@main
+        uses: nimblehq/mulesoft-actions/setup@v1
 
       - name: Run MUnit tests
-        uses: nimblehq/mulesoft-actions/test@main
+        uses: nimblehq/mulesoft-actions/test@v1
         with:
           nexus_username: ${{ secrets.NEXUS_USERNAME }}
           nexus_password: ${{ secrets.NEXUS_PASSWORD }}
           maven_settings_path: .maven/settings.xml
 ```
 
+### Build with Maven
+
+Action to build with Maven. See [build/action.yml](build/action.yml)
+
+#### Usage
+
+```yml
+- uses: nimblehq/mulesoft-actions/build@v1
+  with:
+    # Upload build artifacts to GitHub Actions Artifacts
+    # Default: true
+    use_artifacts: true
+
+    # Artifact name
+    # Default: The name of the build and commit hash
+    artifact_name: build-artifacts
+```
+
+Basic:
+
+```yml
+name: My workflow
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Mulesoft environment
+        uses: nimblehq/mulesoft-actions/setup@v1
+
+      - name: Build with Maven
+        uses: nimblehq/mulesoft-actions/build@v1
+        with:
+          artifact_name: build-artifacts
+```
+
+### Deploy to CloudHub
+
+Action to deploy to CloudHub. See [deploy/action.yml](deploy/action.yml)
+
+> [!IMPORTANT]\
+> This action requires a built artifact from the `build` action.
+> Before deploying to CloudHub, you must configure the `cloudHubDeployment` element. Inside the `org.mule.tools.maven` plugin element in the project’s `pom.xml` file, add the following configuration:
+
+```xml
+<plugin>
+    <groupId>org.mule.tools.maven</groupId>
+    <artifactId>mule-maven-plugin</artifactId>
+    <version>${mule.maven.plugin.version}</version>
+    <extensions>true</extensions>
+    <configuration>
+        <!-- Add the following configuration -->
+        <cloudHubDeployment>
+          <uri>https://anypoint.mulesoft.com</uri>
+          <muleVersion>${app.runtime}</muleVersion>
+          <applicationName>${CLOUDHUB_APPLICATION_NAME}</applicationName>
+          <environment>${CLOUDHUB_ENVIRONMENT}</environment>
+          <businessGroupId>${CLOUDHUB_BUSINESS_GROUP_ID}</businessGroupId>
+          <region>${CLOUDHUB_REGION}</region>
+          <connectedAppClientId>${CONNECTED_APP_CLIENT_ID}</connectedAppClientId>
+          <connectedAppClientSecret>${CONNECTED_APP_CLIENT_SECRET}</connectedAppClientSecret>
+          <connectedAppGrantType>client_credentials</connectedAppGrantType>
+          <properties>
+            <mule.env>${MULE_ENVIRONMENT}</mule.env>
+          </properties>
+        </cloudHubDeployment>
+        <!-- End of configuration -->
+    </configuration>
+</plugin>
+```
+
+#### Usage
+
+```yml
+- uses: nimblehq/mulesoft-actions/deploy@v1
+  with:
+    # Use artifact from `build` action
+    # Default: false
+    use_artifact: true
```

Comment:
is it an option @hoangmirs - or it's mandatory, if it's a mandatory requirement, should we do not make it as optional 🤔